lib/libLog.lua

```diff
@@ -0,0 +1,378 @@
+
+    local result = log.caller(level)
+      level - number - the  number of stack levels to go down to retrieve the caller routine information
+
+  DESCRIPTION
+    caller gets the name of the calling routine and returns it
+
+  RETURN VALUE
+    result - string - the name of the calling function and line number or 'callback: ' if the attempt to get the 
+                      caller returns nil
+
+
+]]
+
+function libLog.caller(level)
+  local name = debug.getinfo(level).name
+  local lineno = nil
+  local source = nil
+    if name then
+      lineno = debug.getinfo(level).currentline
+      -- returns the path to the file prefixed with @
+      source = debug.getinfo(level).source
+      -- we just need the filename, so grab it from the string
+      -- Thanks, Tobias Jakobs  :-)
+      source = string.match(source, "@.-([^\\/]-%.?[^%.\\/]*)$")
+      return name .. ": " .. source .. ": " .. lineno .. ":"
+    else
+      return "callback:"
+    end
+end
+
+libLog.libdoc.functions[#libLog.libdoc.functions + 1] = {
+  Sections = {"Name", "Synopsis", "Description", "Limitations"},
+  Name = "msg - print a log message",
+  Synopsis = [[local log = require "lib/libLog"
+
+    log.msg(level, ...)
+      level - table - the type of message, one of: 
+        libLog.debug   - debugging messages
+        libLog.info    - informational messages
+        libLog.warn    - warning messages 
+        libLog.error   - error messages 
+        libLog.success - success messages
+      ... - string(s) - the message to print, which could be a comma separated set of strings]],
+  Description = [[msg checks the level to see if it is enabled, then prints the level type and message if it is.
+    debug messages print straight to the console, info and warn messages to dt.print_error() and error
+    and success to dt.print()]],
+  Limitations = [[If you use libLog.msg in a callback, the name of the calling routine can't be determined.  A solution
+    is to include some means of reference such as the name of the callback as an argument, i.e. 
+
+      libLog.msg(libLog.debug, "libPlugin.format_combobox:", "value is " .. self.value)
+
+    which would result in
+
+      DEBUG: callback: libPlugin.format_combobox: value is JPEG]],
+}
+
+--[[
+  NAME
+    msg - print a log message
+
+  SYNOPSIS
+    local log = require "lib/libLog"
+
+    log.msg(level, ...)
+      level - table - the type of message, one of: 
+        libLog.debug   - debugging messages
+        libLog.info    - informational messages
+        libLog.warn    - warning messages 
+        libLog.error   - error messages 
+        libLog.success - success messages
+      ... - string(s) - the message to print, which could be a comma separated set of strings
+
+  DESCRIPTION
+    msg checks the level to see if it is enabled, then prints the level type and message if it is.
+    debug messages print straight to the console, info and warn messages to dt.print_error() and error
```

Comment:
That's why I pulled everything back to just console.  When I first thought up the log library, I thought about it writing everything and wrapping dt.print_error() and dt.print().  The more I used it the more I thought my original idea might not have been so good, so I pulled it back.  Then I thought of the dictionary idea so that we could send it where we wanted, but I had too much in my head to do it then.  I like the writing policy.  It gives me an idea where to go with this.